clients/client/src/main/java/org/projectnessie/client/api/CreateReferenceBuilder.java

```diff
@@ -29,11 +31,81 @@
  * @since {@link NessieApiV1}
  */
 public interface CreateReferenceBuilder {
+  /**
+   * Sets the name of the reference that contains the hash of the reference-to-be-created.
+   *
+   * <p>This reference name will be used for authorization purposes to validate that the caller has
+   * read access to the provided hash.
+   *
+   * <p>If not explicitly set, the default branch will be used as the source reference name.
+   *
+   * <p>The name and hash of the reference to be created is set via {@link #reference(Reference)}.
+   *
+   * @param sourceRefName is the name of the reference that contains the hash of the
+   *     reference-to-be-created.
+   */
   CreateReferenceBuilder sourceRefName(
       @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
           String sourceRefName);
 
+  /**
+   * Sets the name and hash of the reference-to-be-created.
+   *
+   * <p>For creating a {@link Tag}, hash cannot be null.
+   *
+   * <p>For creating a {@link Branch}, If the hash is not specified, hash will be set to reference
+   * the beginning of time (aka NO_ANCESTOR hash).
+   *
+   * <p>If the hash is specified, it should be on the {@link #sourceRefName(String)}
+   *
+   * @param reference is {@link Branch} or {@link Tag} defining the name and hash for the new
+   *     reference-to-be-created.
+   */
   CreateReferenceBuilder reference(@Valid @NotNull Reference reference);
 
+  /**
```

Comment:
maybe rather add this to the class itself

Reply:
+1 to moving to class-level javadoc